Frontend/student-hub/src/pages/Profile.js

```diff
@@ -47,10 +46,31 @@ function Profile() {
       } catch (error) {
         setError('Failed to fetch profile.');
       }
-    };
+    }; // fetch unviersity list
+    const fetchUniversity = async () => {
+      try {
+        const { token } = getUserIDToken();
+        const response = await fetch (`${process.env.REACT_APP_API}/profile/university`, {
+          headers: {
+            'Authorization': `Bearer ${token}`
+          }
+        });
+        if (response.ok){
+          const data = await response.json();
+          setUniversities(data)
+        }else{
+          setError(`Failed to fetch universities: ${error.message}`);
```

Comment:
`error` here could be empty or unrelated from a previous error since this isn't catching an error - let's pass in the error message from the response